docker/docker.go

```diff
@@ -28,8 +28,44 @@ import (
 
 type Port = types.Port
 
+var dockerClient *client.Client
+
+// GetClient returns a docker client if all checks pass.
+// This function performs three checks:
+//   1. checks that the user is not running docker toolbox.
+//   2. checks that docker is installed and running properly,
+//   3. checks that the client api version is supported by the docker engine,
+func GetClient() (*client.Client, error) {
+	if dockerClient == nil {
+		logrus.Debug("Checking for Docker Toolbox")
+		// If docker toolbox is installed then the environment variable
+		// `DOCKER_TOOLBOX_INSTALL_PATH` is not empty
```

Comment:
I'm not sure it's a correct way to check for the docker toolbox.
Afaik you can have both toolbox&native installed. But should run only one.

Reply:
> Afaik you can have both toolbox&native installed.

Ouch, didn't know that. Gonna try.

Reply:
Ok I've also installed Docker Toolbox locally and I've fixed the check 👍 [here](https://github.com/src-d/engine/pull/428/commits/370dd8a197f6da8034a02988c19901233add757a).